### DISCUSSION TITLE
[Proposal/Discussion] Merge Mobile and Web

### DISCUSSION THREAD

**Discussion by gaetschwartz** [immich-app/immich#9278] [General]
## Context

Currently, the web interface is using Svelte and the mobile app is using Flutter. Both are great and functional.

## Problem

The fact we are using different frameworks for the two interfaces leads to inevitable problems such as:
- the mobile app lagging behind
- possible incompatibility issues (API endpoint updated in one but not the other)
- less incentives to work on the mobile app, new features have to first be implemented on the web app, less love for the mobile version
- knowledge of 2 different frameworks is required 
- and maybe the most important one, double the work. 

## Proposed solution

Merging the two and using Flutter web as the web interface would help/fix most of the aforementioned problems.

## Feasibility 

I am really not sure about the state of the Svelte app. But from what I had time to explore in the mobile app and codebase one of the only features lacking is the administrator interface. I would gladly volunteer to take up on this task !

## Cons

While Flutter web has matured a lot, it can still be kind of rough around certain edges, but with Flutter’s good evolution pace, it should be only a matter of time before the met pain points get resolved.

**Comment by bo0tzz** [comment]
Previously discussed here: https://github.com/immich-app/immich/issues/181#issuecomment-1136137059

**Reply by gaetschwartz** [reply]
I saw that, but since I it has been two years and the main point of Alex Tran was the immaturity of the framework, I deemed it reasonable to open a new one. Feel free to close it if it’s a duplicate ;)

**Reply by azzamsa** [reply]
Should we ping Alex to hear his opinion on this?

**Reply by alextran1502** [reply]
There will be no merging since each framework excels on its own platform

**Reply by azzamsa** [reply]
> There will be no merging since each framework excels on its own platform

Thanks for taking time to reply.
I love hearing your opinions as I wanted to build an app for both web and mobile. I think I will follow your path. Svelte for web and flutter for mobile.

Thanks a lot for Immich! 🥞

**Reply by alextran1502** [reply]
No problem! Cheers!🎉

**Reply by azzamsa** [reply]
@alextran1502 If you were to develop a desktop version of Immich today, which GUI framework would you choose?

**Reply by alextran1502** [reply]
Electron
